### PR TITLE
attrs 23.1.0

### PR DESCRIPTION
**Pull request by skupr-anaconda**
**Current package info**

| Channel | Downloads | Version | Platforms | License |
| --- | --- | --- | --- | --- |
| defaults | [Conda Downloads](https://anaconda.org/main/attrs) | [Conda Version](https://anaconda.org/main/attrs) | [Conda Platforms](https://anaconda.org/main/attrs) | Conda License |

Changelog: https://github.com/python-attrs/attrs/blob/23.1.0/CHANGELOG.md
License: https://github.com/python-attrs/attrs/blob/23.1.0/LICENSE
Requirements: https://github.com/python-attrs/attrs/blob/23.1.0/pyproject.toml

Actions:
1. Add a patch to remove `fancy-pypi-readme` dependency
2. Clean up the recipe
3. Add `--no-build-isolation` to script
4. Update host: use hatchling, hatch-vcs

### CODE REVIEW

File: recipe/patches/001-remove-fancy-pypi-readme.patch [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+diff --git a/pyproject.toml b/pyproject.toml
+index fb8fae3..010357e 100644
+--- a/pyproject.toml
++++ b/pyproject.toml
+@@ -1,7 +1,7 @@
+ # SPDX-License-Identifier: MIT
+ 
+ [build-system]
+-requires = ["hatchling", "hatch-vcs", "hatch-fancy-pypi-readme"]
```
Comment (M-Waszkiewicz-Anaconda) [inline recipe/patches/001-remove-fancy-pypi-readme.patch:18]: instead of removing that dependency why we cannot add this as a dependent package ?

Comment (skupriienko) [reply recipe/patches/001-remove-fancy-pypi-readme.patch:18]: It's not needed for building a package.  jsonschema and black packages have the same approach

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,26 +1,26 @@
...
 requirements:
   host:
     - pip
     - python
-    - setuptools
-    - wheel
+    - hatchling
+    - hatch-vcs
   run:
     - python
```
Comment (akabanovs) [inline recipe/meta.yaml:29]: Do we want to add functional tests that seem to be available, or it is safe to skip those?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,26 +1,26 @@
...
+  patches:
+    - patches/001-remove-fancy-pypi-readme.patch
 
 build:
   number: 0
-  script: python -m pip install . --no-deps --ignore-installed
+  script: python -m pip install . --no-deps --no-build-isolation -vv
 
 requirements:
   host:
```
Comment (akabanovs) [inline recipe/meta.yaml:22]: Do we need to add patch as a build requirement?

Comment (skupr-anaconda) [reply recipe/meta.yaml:22]: Fixed

File: recipe/meta.yaml (hunk 2 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:29]: I've added a test suite

**Review by akabanovs — APPROVED** [review]

File: recipe/patches/001-remove-fancy-pypi-readme.patch (hunk 1 above)
Comment (JeanChristopheMorinPerso) [reply recipe/patches/001-remove-fancy-pypi-readme.patch:18]: Indeed, `hatch-fancy-pypi-readme` is only required to generated the README in the generated wheel metadata (https://inspector.pypi.io/project/attrs/23.1.0/packages/f0/eb/fcb708c7bf5056045e9e98f62b93bd7467eb718b0202e7698eb11d66416c/attrs-23.1.0-py3-none-any.whl/attrs-23.1.0.dist-info/METADATA#line.53).

**Review by JeanChristopheMorinPerso — APPROVED** [review]

**Pull request merged**